webapp/tests/test_events.py

```diff
@@ -87,6 +87,57 @@ def test_tag_as_str(self):
         self.assertEqual(error, {'error': '"tags" must be an array'})
         self.assertEqual(Event.objects.count(), 0)
 
+    def test_tag_sets(self):
+        creation_url = reverse('graphite.events.views.view_events')
+        events = [
+            {
+                'what': 'Something happened',
+                'data': 'more info',
+                'tags': ['foo'],
+            },
+            {
+                'what': 'Something else happened',
+                'data': 'even more info',
+                'tags': ['bar'],
+            },
+            {
+                'what': 'A final thing happened',
+                'data': 'yet even more info',
+                'tags': ['foo', 'bar'],
+            },
+        ]
+
+        for event in events:
+            response = self.client.post(creation_url, json.dumps(event),
+                                        content_type='application/json')
+            self.assertEqual(response.status_code, 200)
+
+        url = reverse('events_get_data')
+
+        # should match two events using old set logic
+        response = self.client.get(url, {'tags': 'foo'})
+        self.assertEqual(response.status_code, 200)
+        events = json.loads(response.content)
+        self.assertEqual(len(events), 2)
+
+        # should match one event using set intersection
+        response = self.client.get("%s%s" % (url, '?tags=foo%20bar&set=intersection'))
+        self.assertEqual(response.status_code, 200)
+        events = json.loads(response.content)
+        self.assertEqual(len(events), 1)
+
+        # should match zero events using set intersection and unknown tag
+        response = self.client.get("%s%s" % (url, '?tags=foo%20bar%20nope&set=intersection'))
```

Comment:
@cbowman0 Any idea why this keeps returning one match instead of zero? The `curl` equivalent of this test works as expected so I'm not sure why it keeps returning an unwanted result.